haskell/private/pkgdb_to_bzl.py

```diff
@@ -0,0 +1,170 @@
+                    id = "{id}",
+                    deps = {deps},
+                    hdrs = {hdrs},
+                    includes = {includes},
+                    linkopts = {linkopts},
+                    shared_library = {shared_library}[0] if {shared_library} else None,
+                    static_library = {static_library}[0] if {static_library} else None,
+                    static_profiling_library = {static_profiling_library}[0] if {static_profiling_library} else None,
+                    version = "{version}",
+                    visibility = ["//visibility:public"],
+                )
+                """.format(
+                    name = pkg.name,
+                    id = pkg.id,
+                    version = pkg.version,
+                    hdrs = "glob({})".format([
+                        path_to_label("{}/**/*.h".format(include_dir), pkg.pkgroot)
+                        for include_dir in pkg.include_dirs
+                        if path_to_label(include_dir, pkg.pkgroot)
+                    ]),
+                    includes = [
+                        "/".join([repo_dir, path_to_label(include_dir, pkg.pkgroot)])
+                        for include_dir in pkg.include_dirs
+                        if path_to_label(include_dir, pkg.pkgroot)
+                    ],
+                    static_library = "glob({})".format([
+                        path_to_label("{}/lib{}.a".format(library_dir, hs_library), pkg.pkgroot)
+                        for library_dir in pkg.library_dirs
+                        if path_to_label(library_dir, pkg.pkgroot)
+                    ]),
+                    static_profiling_library = "glob({})".format([
+                        path_to_label("{}/lib{}_p.a".format(library_dir, hs_library), pkg.pkgroot)
+                        for library_dir in pkg.library_dirs
+                        if path_to_label(library_dir, pkg.pkgroot)
+                    ]),
+                    shared_library = "glob({})".format([
+                        path_to_label(
+                            "{}/lib{}-ghc*.{}".format(
+                                dynamic_library_dir,
+                                hs_library,
+                                ext,
+                            ),
+                            pkg.pkgroot,
+                        )
+                        for dynamic_library_dir in pkg.dynamic_library_dirs + pkg.library_dirs
+                        if path_to_label(dynamic_library_dir, pkg.pkgroot)
+                        for ext in ["dll", "dylib", "so"]
+                    ]),
+                    deps = pkg.depends,
+                    linkopts = pkg.ld_options + [
+                        "-L{}".format(library_dir)
+                        for library_dir in pkg.dynamic_library_dirs + pkg.library_dirs
+                        if not path_to_label(library_dir, pkg.pkgroot)
+                    ] + [
+                        "-l{}".format(extra_library)
```

Comment:
If `extra_library` is a static library, do these `linkopts` end up in the right position in the linker flags? I.e. behind the object that depends on the `extra_library`.

Reply:
Needs a comment.